tensorflow/python/lib/io/file_io.py

```diff
@@ -48,7 +49,7 @@ class FileIO(object):
   """
 
   def __init__(self, name, mode):
-    self.__name = name
+    self.__name = compat.path_to_str(name)
```

Comment:
Alternatively I could make the conversion directly before `self.__name` is passed to file system layer which has the potential benefit that `FileIO.name` would return the original pathlike object without conversion. Though I am not 100% sure what the right behaviour there should be.

Reply:
Let's leave it here unless we need the invariant that `FileIO(path, mode).name == path` from the outside of the class.

Having it here ensures that this class is consistent with the requirements at the C/C++ API layer.

Reply:
Actually, now that I read the entire PR, we can move it to just before passing to the filesystem layer as we already do other similar conversions. Let's do that.

Reply:
I agree, I only realized that this would be a good option after implementing the changes for the other `tf.io.gfile.*` functions. I changed it in 5ef85fc6c826501aa852aed8228ec5dd8beeb07c